### ISSUE TITLE
Error loading C4 realnewslike dataset

### ISSUE THREAD

**Issue by danshirron** [huggingface/datasets#2761]
## Describe the bug
Error loading C4 realnewslike dataset. Validation part mismatch

## Steps to reproduce the bug
```python
 raw_datasets = load_dataset('c4', 'realnewslike', cache_dir=model_args.cache_dir)
## Expected results
success on data loading
## Actual results
Downloading: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 15.3M/15.3M [00:00<00:00, 28.1MB/s]Traceback (most recent call last):                                                                                                                                                                                                                                             
  File "run_mlm_tf.py", line 794, in <module>                                                                                                                                                                                                                                  
    main()                                                                                                                                                                                                                                                                     
  File "run_mlm_tf.py", line 425, in main                                                                                                                                                                                                                                      
    raw_datasets = load_dataset(data_args.dataset_name, data_args.dataset_config_name, cache_dir=model_args.cache_dir)                                                                                                                                                           File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/load.py", line 843, in load_dataset                                                                                                                                                                         
    builder_instance.download_and_prepare(                                                                                                                                                                                                                                     
  File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/builder.py", line 608, in download_and_prepare                                                                                                                                                              
    self._download_and_prepare(                                                                                                                                                                                                                                                
  File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/builder.py", line 698, in _download_and_prepare                                                                                                                                                                 verify_splits(self.info.splits, split_dict)                                                                                                                                                                                                                                  File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/utils/info_utils.py", line 74, in verify_splits                                                                                                                                                             
    raise NonMatchingSplitsSizesError(str(bad_splits))                                                                                                                                                                                                                         
datasets.utils.info_utils.NonMatchingSplitsSizesError: [{'expected': SplitInfo(name='validation', num_bytes=38165657946, num_examples=13799838, dataset_name='c4'), 'recorded': SplitInfo(name='validation', num_bytes=37875873, num_examples=13863, dataset_name='c4')}] 

## Environment info
- `datasets` version: 1.10.2
- Platform: Linux-5.4.0-58-generic-x86_64-with-glibc2.29
- Python version: 3.8.10
- PyArrow version: 4.0.1

**Comment by bhavitvyamalik** [comment]
Hi @danshirron, 
`c4` was updated few days back by @lhoestq. The new configs are `['en', 'en.noclean', 'en.realnewslike', 'en.webtextlike'].` You'll need to remove any older version of this dataset you previously downloaded and then run `load_dataset` again with new configuration.

**Comment by danshirron** [comment]
@bhavitvyamalik @lhoestq , just tried the above and got:
>>> a=datasets.load_dataset('c4','en.realnewslike')
Downloading: 3.29kB [00:00, 1.66MB/s]                                                                                                                                                                                                                                          
Downloading: 2.40MB [00:00, 12.6MB/s]                                                                                                                                                                                                                                          
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/load.py", line 819, in load_dataset
    builder_instance = load_dataset_builder(
  File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/load.py", line 701, in load_dataset_builder
    builder_instance: DatasetBuilder = builder_cls(
  File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/builder.py", line 1049, in __init__
    super(GeneratorBasedBuilder, self).__init__(*args, **kwargs)
  File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/builder.py", line 268, in __init__
    self.config, self.config_id = self._create_builder_config(
  File "/home/dshirron/.local/lib/python3.8/site-packages/datasets/builder.py", line 360, in _create_builder_config
    raise ValueError(
ValueError: BuilderConfig en.realnewslike not found. Available: ['en', 'realnewslike', 'en.noblocklist', 'en.noclean']
>>> 

datasets version is 1.11.0

**Comment by bhavitvyamalik** [comment]
I think I had the older version of datasets installed and that's why I showed the old configurations to you in my last comment, my bad! I re-checked and updated it to latest version (`datasets==1.11.0`) and it's showing `available configs: ['en', 'realnewslike', 'en.noblocklist', 'en.noclean']`. I tried `raw_datasets = load_dataset('c4', 'realnewslike')` but due to insufficient disk space it couldn't download. Clearing the cache and using one of  `available configs: ['en', 'realnewslike', 'en.noblocklist', 'en.noclean']` should work.

**Comment by bhavitvyamalik** [comment]
I think I had an older version of datasets installed and that's why I commented the old configurations in my last comment, my bad! I re-checked and updated it to latest version (`datasets==1.11.0`) and it's showing `available configs: ['en', 'realnewslike', 'en.noblocklist', 'en.noclean']`. 

I tried `raw_datasets = load_dataset('c4', 'realnewslike')` and the download started. Make sure you don't have any old copy of this dataset and you download it fresh using the latest version of datasets. Sorry for the mix up!
